### DISCUSSION TITLE
Reindex concurrently using pg_timetable

### DISCUSSION THREAD

**Discussion by ArashMasroor** [cybertec-postgresql/pg_timetable#659] [Q&A]
I am using pg_timetable to  reindex concurrently some of my indexes.
While doing the reindex, another query comes and locks the table. At this time an error appears and the command does not get finished.

Here is my config:

![1](https://github.com/user-attachments/assets/9b97b1e7-8ad2-45db-a203-294f3746157e)
![2](https://github.com/user-attachments/assets/d6a2ee74-e779-46e3-a335-e6e4068be044)
![3](https://github.com/user-attachments/assets/1d8b451e-ffc3-482c-90ee-897998720f58)

Here is the execution log:
![4](https://github.com/user-attachments/assets/c598f86a-349a-47c5-abbb-fe1900d7fe73)

Here is the timetable log:
![5](https://github.com/user-attachments/assets/b58f51f8-d801-4a6a-b458-3b8ff198737b)

How is it possible to overcome this error?

**Comment by pashagolub** [comment]
Would you please provide the full error message for `unexpected EOF`.

Seems like you have connection issues and it's nothing we can do on pg_timetable side.

**Comment by ArashMasroor** [comment]
There is not any error other than the logs I posted. If there is, please guide me to get it then I will post it.

As I explained the reason is the other queries that lock the table,which I am reindexing by the job.I wish the job could wait until they finish and unlock the table.
May be if the time out get increased, the problem would be solved.

Here the time out field is set to 0,but job gets failed as soon as the lock occures.

**Comment by pashagolub** [comment]
There are such things as [statement_timeout](https://postgresqlco.nf/doc/en/param/statement_timeout/), [idle_in_transaction_session_timeout](https://postgresqlco.nf/doc/en/param/idle_in_transaction_session_timeout/) and [idle_session_timeout](https://postgresqlco.nf/doc/en/param/idle_session_timeout/). And these options are controlled by you. Please make sure the session is not dropped by the server due these settings.

pg_timetable has timeouts for chains and tasks. Default value 0 means wait forever.

**Comment by ArashMasroor** [comment]
Sorry for the delay.

This is the value of those parameters:
**idle_in_transaction_session_timeout** is set to	**600000**
**statement_timeout**	is set to **0**
**idle_session_timeout** is not set at all and is not in the .config file.
